kernels/volk/volk_8u_conv_k7_r2puppet_8u.h

```diff
@@ -20,11 +20,14 @@ typedef union {
     unsigned int* w;
 } p_decision_t;
 
-static inline int parity(int x, unsigned char* Partab)
+static inline int parity(int x)
 {
-    x ^= (x >> 16);
-    x ^= (x >> 8);
-    return Partab[x];
+    x ^= x >> 16;
+    x ^= x >> 8;
+    x ^= x >> 4;
+    x ^= x >> 2;
+    x ^= x >> 1;
+    return x & 1;
 }
```

Comment:
Could you give a quick explanation or reference for this change? I like it. It's for documentation purposes.

Reply:
I just noticed that the kernels are doing a lot of redundant work.
To calculate the parity of 64 numbers, the code first fills a lookup table of 256 numbers with the result of a parity calculation to optimize the parity calculation of the 64 numbers in the same codepath.

Reply:
I'm a bit worried because we might break a test in an already fragile kernel. Still, it makes sense, let's go with your solution.